dist/apache-unomi-tracker.cjs.js

```diff
@@ -0,0 +1,1571 @@
+
+        var selectedFilter = null;
+
+        if (successfulFilters.length > 0) {
+          selectedFilter = successfulFilters[0];
+          var minPos = successfulFilters[0].position;
+
+          if (minPos >= 0) {
+            for (var j = 1; j < successfulFilters.length; j++) {
+              if (successfulFilters[j].position < minPos) {
+                selectedFilter = successfulFilters[j];
+              }
+            }
+          }
+        }
+
+        if (resultCallback) {
+          // execute callback
+          resultCallback(successfulFilters, selectedFilter);
+        } else {
+          if (selectedFilter) {
+            var targetFilters = document.getElementById(target).children;
+
+            for (var fIndex in targetFilters) {
+              var filter = targetFilters.item(fIndex);
+
+              if (filter) {
+                filter.style.display = filter.id === selectedFilter.content ? '' : 'none';
+              }
+            } // we now add control group information to event if the user is in the control group.
+
+
+            if (wem._isInControlGroup(target)) {
+              console.info('[WEM] Profile is in control group for target: ' + target + ', adding to personalization event...');
+              selectedFilter.event.target.properties.inControlGroup = true;
+
+              if (selectedFilter.event.target.properties.variants) {
+                selectedFilter.event.target.properties.variants.forEach(function (variant) {
+                  return variant.inControlGroup = true;
+                });
+              }
+            } // send event to unomi
+
+
+            wem.collectEvent(wem._completeEvent(selectedFilter.event), function () {
+              console.info('[WEM] Personalization event successfully collected.');
+            }, function () {
+              console.error('[WEM] Could not send personalization event.');
+            }); //Trigger variant display event for personalization
+
+            wem._dispatchJSExperienceDisplayedEvent(selectedFilter.event);
+          } else {
+            var elements = document.getElementById(target).children;
+
+            for (var eIndex in elements) {
+              var el = elements.item(eIndex);
+              el.style.display = 'none';
+            }
+          }
+        }
+      });
+    },
+
+    /**
+     * This function will register an optimization test or A/B test
+     *
+     * @param {string} optimizationTestNodeId the optimization test node id
+     * @param {string} goalId the associated goal Id
+     * @param {string} containerId the HTML container Id
+     * @param {object} variants the variants
+     * @param {boolean} [ajax] Deprecated: Ajax rendering is not supported anymore
+     * @param {object} [variantsTraffic] the associated traffic allocation
+     * @return {undefined}
+     */
+    registerOptimizationTest: function registerOptimizationTest(optimizationTestNodeId, goalId, containerId, variants, ajax, variantsTraffic) {
```

Comment:
goalId, containerId and ajax are not used by the function do we want to keep them?

Reply:
I prefer not for now, because it would break any existing call to this function.
In our own project, we may still use it with this param, and removing it here would:
- either break our code
- either force us to redefined the signature on our side
This definitely should be done, but again the code needed to be outsourced at it is for a first commit.

Then discussion will be open to see what to improve and clean